website/utils/url_utils.ts

```diff
@@ -2,16 +2,56 @@ import FirebaseRequest, { DatasetMetaData } from "../../public/firebase";
 
 import { AppProps, GlobalViewerSettings } from "../../src/aics-image-viewer/components/App/types";
 import { ViewMode } from "../../src/aics-image-viewer/shared/enums";
-import { ViewerChannelSettings } from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
+import {
+  ViewerChannelSetting,
+  ViewerChannelSettings,
+} from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
 
-const paramKeys = ["mask", "ch", "luts", "colors", "url", "file", "dataset", "id", "view"];
-type ParamKeys = (typeof paramKeys)[number];
-type Params = { [_ in ParamKeys]?: string };
+const CHANNEL_STATE_KEY_REGEX = /^c[0-9]+$/;
+// Match colon-separated pairs of alphanumeric strings
+const LUT_REGEX = /^[a-z0-9.]*:[ ]*[a-z0-9.]*$/;
+const HEX_COLOR_REGEX = /^[0-9a-fA-F]{6}$/;
 
-function urlSearchParamsToParams(searchParams: URLSearchParams): Params {
+export type ViewerChannelSettingJson = {
+  /** Color, as a 6-digit hex color */
+  col?: string;
+  /** Colorize. "1" is enabled. */
+  clz?: "1" | "0";
+  /** Colorize alpha, in the [0, 1] range. */
+  cza?: string;
+  /** Opacity, in the [0, 1 range] */
+  isa?: string;
+  /** LUT to map from intensity to opacity. Should be two alphanumeric values separated
+   * by a colon, e.g. "0:255", "p50:p90", or "autoij:0".
```

Comment:
Is this the right place to describe/document the encodings?  Plain number = intensity, p[Number 0-100] = intensity of percentile, m[Number]=median intensity * number, autoij = override with algorithm to automatically select min and max.

Reply:
For documentation purposes, it would be good to know what the defaults are for everything too. In one place.

Reply:
Included, thanks!